src/1_utils.js

```diff
@@ -61,6 +61,42 @@ utils.userPreferences = {
 	}
 };
 
+utils.generateDynamicBNCLink = function(branchKey, data) {
+	if (!branchKey && !data) {
+		return;
+	}
+	if (!data["data"] || typeof data["data"] !== "string") {
+		data = utils.cleanLinkData(data);
+	}
+	var addKeyAndValueToUrl = function(fallbackUrl, tagName, tagData) {
+		var first = fallbackUrl[fallbackUrl.length - 1] === "?";
+		var modifiedFallbackURL = first ? fallbackUrl + tagName : fallbackUrl + "&" + tagName;
+		modifiedFallbackURL += "=";
+		return tagName === "data" ? modifiedFallbackURL + encodeURIComponent(utils.base64encode(tagData)) : modifiedFallbackURL + encodeURIComponent(tagData);
+	};
+
+	var fallbackUrl = config.link_service_endpoint + '/a/' + branchKey + '?';
+	var topLevelKeys = [ "tags", "alias", "channel", "feature", "stage", "campaign", "type", "duration", "sdk", "source", "data" ];
+	for (var i = 0; i < topLevelKeys.length; i++) {
+		var key = topLevelKeys[i];
+		var value = data[key];
+		if (value) {
+			if (key === "tags" && Array.isArray(value)) {
+				for (var index = 0; index < value.length; index++) {
+					fallbackUrl = addKeyAndValueToUrl(fallbackUrl, key, value[index]);
```

Comment:
if you have multiple tags, won't it end up looking like this: `http://fallbackurl.com/?tags=tag1&tags=tag2&tags=tag3`? I don't think this will parse properly